docs/osf/addons.rst

```diff
@@ -31,31 +32,218 @@ During your installation you created a virtual environment for OSF. Switch to th
+    │   ├── node-cfg.js*
+    │   ├── tests
+    │   │   └── ...
+    │   └── user-cfg.js*
+    ├── templates
+    │   ├── log_templates.mako
+    │   ├── addonshortname_node_settings.mako*
+    │   └── addonshortname_user_settings.mako*
+    ├── tests
+    │   ├── __init__.py
+    │   ├── test_model.py
+    │   └── test_views.py
+    └── views
+        └── ...
+
+\* optional
+
+StoredObject
+************
+
+All models should be defined as subclasses of `framework.mongo.StoredObject`.
+
+
+Routes
+******
+
+Routes are defined in a dictionary containing ``rules`` and an optional ``prefix``.
+
+Our url templating works the same way that `flask’s`_ does.
+
+.. code:: python
+
+    my_route = {
+      'rules': [
+        Rule(
+          [
+            '/my/<templated>/path/',  # Note all routes SHOULD end with a forward slash (/)
+            '/also/my/<templated>/path/'
+          ],
+          ('get', 'post'),  # Valid HTTP methods
+          view.my_view_function,  # The view method this route maps to
+          json_renderer  # The renderer used for this view function, either OsfWebRenderer or json_renderer
+        )
+      ]
+    }
+
+Routes SHOULD be defined in ``website.addons.youraddon.routes`` but
+could be defined anywhere
+
+Views
+*****
 
-This will prompt a few questions:
+Our views are implemented the same way that `flask’s`_ are.
 
-**full_name (default is "Amazon Simple Storage Service")?**: Enter a descriptive name that can have spaces. Describe your add on in a few words.
+Any value matched by url templating (``<value_name>``) will be passed to
+your view function as a keyword argument
 
-**short_name (default is "s3")?**: Enter a lowercase, single word name. This name will be used in the file structure and as variables. You can use CamelCase if you need to have more words, or you can combine with numbers, but don't use any spaces or hyphens.
+Our framework supplies many python decorators to make writing view
+functions more pleasant.
 
-**categories (default is "'storage'")?**: Enter a one word category, lowercase.
+Below are a few examples that are commonly used in our code base.
 
+More can be found in ``website.project.decorators``.
+
+``framework.auth.decorators.must_be_logged_in``
+-----------------------------------------------
+
+Ensures that a user is logged in and imputes ``auth`` into keyword
+arguments
+
+``from website.project.decorators.must_have_addon``
+---------------------------------------------------
+
+``must_have_addon`` is a decorator factory meaning you must supply
+arguments to it to get a decorator.
+
+.. code:: python
+
+    @must_have_addon('myaddon', 'user')
+    def my_view(...):
+      pass
+
+
+    @must_have_addon('myaddon', 'node')
+    def my_node_view(...):
+      pass
+
+The above code snippet will only run the view function if the specified
+model as the requested addon.
+
+.. note:: 
+    Routes whose views are with decorated ``must_have_addon`` MUST start with ``/project/<pid>/...``.
```

Comment:
This is only true if the second argument is `node`. Wouldn't hurt to be explicit with `must_have_addon('addon_name', 'node')`.